src/exception.c

```diff
@@ -36,6 +40,145 @@ void register_exception_handler( void (*cb)(exception_t*))
 	__exception_handler = cb;
 }
 
+/**
+ * We keep a function outside the exception handler so that
+ * the interrupts are re-enabled when we return from the ex.
+ * handler. This will make sure latest console contents are
+ * displayed.
+ */
+static void exception_halt() {
+	while(1) {
+		console_render();
+	}
+}
+
+void exception_default_handler(exception_t* ex) {
+	// We will naturally return from the ex handler, which
+	// will gracefully restore interrupt state so that we can
+	// continue using interrupt dependent functionalities
+	// such as console updates.
+	C0_WRITE_EPC(&exception_halt);
```

Comment:
Ill try that but I suspect it wont work

Reply:
Unfortunately it does not work. I tried to comment on why it does not work without the interrupts. It may not be 100% accurate and I think it is not worth pursuing further just to remove the extra function. We may have a look again when we update he display system someday :)